igor-web/src/main/java/com/netflix/spinnaker/igor/concourse/ConcourseBuildMonitor.java

```diff
@@ -124,8 +124,13 @@ private JobDelta jobDelta(ConcourseProperties.Host host, Job job) {
     List<GenericBuild> genericBuilds =
         builds.stream()
             .map(build -> concourseService.getGenericBuild(jobPath, build, false))
+            .filter(b -> !cache.getEventPosted(host, job, cursor, b.getNumber()))
```

Comment:
We still want to run `cache.setLastPollCycleTimestamp()` for jobs without uncommitted successful builds, which filtering here prevents.

Reply:
Ah okay. I should be able to add a `cache.setLastPollCycleTimestamp()` in the `if` right after this then right?  

I believe we need the filter + return null there when there are no uncommitted successful builds or else the JobPollingDelta list will always be the # of concourse jobs, correct?  
Just want to make sure I'm understanding the problem.